src/rwe/Unit.cpp

```diff
@@ -108,6 +113,15 @@ namespace rwe
         return BuildCostInfo{buildTimeContribution, deltaEnergy, deltaMetal};
     }
 
+    bool Unit::addRepairProgress(unsigned int repairTimeContribution)
+    {
+        if (buildTimeCompleted == buildTime)
+        {
+            buildTimeCompleted = (hitPoints * buildTime) / maxHitPoints;
+        }
```

Comment:
Not really sure I'm convinced by this (and the addition of `LifeState::Spawning`) -- repairing shouldn't care about buildtime at all.